pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -23,101 +26,127 @@ type DirectApplier struct {
+                case "--force":
+                        applyOpts.ForceConflicts = true
+                case "--prune":
+                        applyOpts.Prune = true
+                case "--selector":
+                        applyOpts.Selector = opt.ExtraArgs[i+1]
+                default:
+                        continue
+                }
+        }
+
+        applyOpts.Namespace = opt.Namespace
+        applyOpts.SetObjects(infos)
+        applyOpts.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
+                applyOpts.PrintFlags.NamePrintFlags.Operation = operation
+                cmdutil.PrintFlagsWithDryRunStrategy(applyOpts.PrintFlags, applyOpts.DryRunStrategy)
+                return applyOpts.PrintFlags.ToPrinter()
+        }
+        applyOpts.DeleteOptions = &cmdDelete.DeleteOptions{
+                IOStreams: ioStreams,
+        }
+
+        return applyOpts.Run()
 }
 
 // staticRESTClientGetter returns a fixed RESTClient
 type staticRESTClientGetter struct {
-	RESTConfig      *rest.Config
-	DiscoveryClient discovery.CachedDiscoveryInterface
-	RESTMapper      meta.RESTMapper
+        RESTConfig      *rest.Config
+        DiscoveryClient discovery.CachedDiscoveryInterface
+        RESTMapper      meta.RESTMapper
 }
 
 var _ resource.RESTClientGetter = &staticRESTClientGetter{}
 
 func (s *staticRESTClientGetter) ToRESTConfig() (*rest.Config, error) {
-	if s.RESTConfig == nil {
-		return nil, fmt.Errorf("RESTConfig not set")
-	}
-	return s.RESTConfig, nil
+        if s.RESTConfig == nil {
+                return nil, fmt.Errorf("RESTConfig not set")
+        }
+        return s.RESTConfig, nil
 }
 func (s *staticRESTClientGetter) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
-	if s.DiscoveryClient == nil {
-		return nil, fmt.Errorf("DiscoveryClient not set")
-	}
-	return s.DiscoveryClient, nil
+        if s.DiscoveryClient == nil {
+                return nil, fmt.Errorf("DiscoveryClient not set")
+        }
+        return s.DiscoveryClient, nil
 }
 func (s *staticRESTClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
-	if s.RESTMapper == nil {
-		return nil, fmt.Errorf("RESTMapper not set")
-	}
-	return s.RESTMapper, nil
+        if s.RESTMapper == nil {
+                return nil, fmt.Errorf("RESTMapper not set")
+        }
+        return s.RESTMapper, nil
+}
+
+// NewApplyOptions creates new ApplyOptions for the `apply` command
+func NewApplyOptions(ioStreams genericclioptions.IOStreams) *apply.ApplyOptions {
+        return &apply.ApplyOptions{
+                PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+
+                Overwrite:    true,
+                OpenAPIPatch: true,
+
+                Recorder: genericclioptions.NoopRecorder{},
+
+                IOStreams: ioStreams,
+
+                VisitedUids:       sets.NewString(),
+                VisitedNamespaces: sets.NewString(),
+        }
 }
```

Comment:
I would remove this `NewApplyOptions` function since we have a better implementation that goes along with the new changes in the k8s 1.24 api changes